checkout.js

```diff
@@ -1,5 +1,7 @@
 function selectSaleItems(items) {
-  return items
+  return items.filter(function discount(items) {
```

Comment:
No harm in naming this function discounts but remember it is being passed in directly and never usable anywhere else so an anonymous function is fine and is actually best practice.